Cross-compilation support

I've tried to cross-compile telescope and build failed with:
```
telescope-aarch64-unknown-linux-gnu> aarch64-unknown-linux-gnu-gcc  -I/nix/store/2dvz4kwky6qxx75qlwjawa52ixpi0lb5-libevent-aarch64-unknown-linux-gnu-2.1.12-dev/include -g -O2 -Wall -Wextra -Wmissing-prototypes -Wstrict-prototypes -Wwrite-strings -Wno-unused-parameter   -o pagebundler pagebundler.o compat/freezero.o compat/getdtablecount.o compat/getprogname.o compat/recallocarray.o compat/strlcat.o compat/strlc
py.o compat/strtonum.o compat/setproctitle.o compat/imsg.o compat/imsg-buffer.o compat/ohash.o compat/fmt_scaled.o -L/nix/store/mq7pij264yrh1piv55a4z5wzxrsyjc8b-libevent-aarch64-unknown-linux-gnu-2.1.12/lib -levent_core -ltls -lncursesw
telescope-aarch64-unknown-linux-gnu> echo > pages.c
telescope-aarch64-unknown-linux-gnu> echo "#include \"pages.h\"" >> pages.c
telescope-aarch64-unknown-linux-gnu> ./pagebundler -f ./pages/about_about.gmi   -v about_about   >> pages.c
telescope-aarch64-unknown-linux-gnu> /nix/store/l0wlqpbsvh1pgvhcdhw7qkka3d31si7k-bash-5.1-p8/bin/bash: line 1: ./pagebundler: cannot execute binary file: Exec format error
telescope-aarch64-unknown-linux-gnu> make: *** [Makefile:1181: pages.c] Error 126
error: builder for '/nix/store/n9ghpkkns8yfbr4xjgjv4y9dr9w759d8-telescope-aarch64-unknown-linux-gnu-0.5.2.drv' failed with exit code 2;
       last 10 log lines:
       >    72 |        __glibc_objsize (__s), __fmt,
       >       |        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
       >    73 |        __va_arg_pack ());
       >       |        ~~~~~~~~~~~~~~~~~
       > aarch64-unknown-linux-gnu-gcc  -I/nix/store/2dvz4kwky6qxx75qlwjawa52ixpi0lb5-libevent-aarch64-unknown-linux-gnu-2.1.12-dev/include -g -O2 -Wall -Wextra -Wmissing-prototypes -Wstrict-prototypes -Wwrite-strings -Wno-unused-parameter   -o pagebundler pagebundler.o compat/freezero.o compat/getdtablecount.o compat/getprogname.o compat/recallocarray.o compat/strlcat.o compat/strlcpy.o compat/strtonum.o compat/setproctitle.o compat/imsg.o compat/imsg-buffer.o compat/ohash.o compat/fmt_scaled.o -L/nix/store/mq7pij264yrh1piv55a4z5wzxrsyjc8b-libevent-aarch64-unknown-linux-gnu-2.1.12/lib -levent_core -ltls -lncursesw
       > echo > pages.c
       > echo "#include \"pages.h\"" >> pages.c
       > ./pagebundler -f ./pages/about_about.gmi   -v about_about   >> pages.c
       > /nix/store/l0wlqpbsvh1pgvhcdhw7qkka3d31si7k-bash-5.1-p8/bin/bash: line 1: ./pagebundler: cannot execute binary file: Exec format error
       > make: *** [Makefile:1181: pages.c] Error 126
```
The reason is that **pagebundler** should be built with build-platform compiler, not with target-pllatform compiler.

--- a/Makefile.am
+++ b/Makefile.am
@@ -52,6 +52,9 @@ telescope_SOURCES +=	phos/phos.h	\
 noinst_PROGRAMS =	pagebundler
 pagebundler_SOURCES =	pagebundler.c
 
+pagebundler$(EXEEXT): pagebundler.c
+	$(HOSTCC) $(HOSTCFLAGS) -o $@ pagebundler.c
+
 BUILT_SOURCES =		cmd.gen.c compile_flags.txt emoji-matcher.c pages.c
 
 CLEANFILES =		cmd.gen.c compile_flags.txt emoji-matcher.c pages.c \

--- a/configure.ac
+++ b/configure.ac
@@ -8,6 +8,12 @@ AC_PROG_YACC
 
 PKG_PROG_PKG_CONFIG
 
+AC_ARG_VAR(HOSTCC, [The C compiler on the host.])
+AC_ARG_VAR(HOSTCFLAGS, [CFLAGS for the host compiler])
+
+test -z "${HOSTCC}"     && HOSTCC='$(CC)'
+test -z "${HOSTCFLAGS}" && HOSTCFLAGS='$(CFLAGS)'
+
 AC_ARG_WITH([libimsg],
 	AS_HELP_STRING([--with-libimsg],
 		[Build with imsg-compat library (default: disabled)]))

